In [1]:
import numpy as np
import pandas as pd

In [2]:
#test dataset
data = pd.read_json("data/RS_2014-12",lines = True)
data.head()

retrieved_on  ups media                                           selftext  \
0    1441043920    1  None                                                      
1    1441043920    2  None  I'd like to submit a childhood favourite of mi...   
2    1441043920    3  None  So my gf wants to play old school nintendo gam...   
3    1441043920    0  None  Why the fuck dreamhack includes players in str...   
4    1441043920   11  None                                                      

                                           permalink link_flair_text  \
0  /r/news/comments/2nw07t/i_hope_you_die_of_aids...            None   
1  /r/AskReddit/comments/2nw07u/its_turned_decemb...            None   
2  /r/pcmasterrace/comments/2nw07v/playstation_co...            None   
3     /r/starcraft/comments/2nw07w/thanks_dreamhack/         [Other]   
4  /r/news/comments/2nw07x/ohio_state_football_pl...            None   

  secure_media_embed                                              title  \
0                 {}                  I hope you die of AIDS in prison.   
1                 {}  It's turned December 1st here in the UK; Reddi...   
2                 {}      Playstation controllers not being recognized?   
3                 {}                                   Thanks dreamhack   
4                 {}  Ohio State Football Player Kosta Karageorge Fo...   

  distinguished                                          thumbnail  ...  \
0          None                                            default  ...   
1          None                                               self  ...   
2          None                                               self  ...   
3          None                                               self  ...   
4          None  http://a.thumbs.redditmedia.com/9NAX6O7TsxpsDt...  ...   

   adserver_click_url  href_url third_party_tracking_2 mobile_ad_url  \
0                 NaN       NaN                    NaN           NaN   
1                 NaN       NaN                    NaN           NaN   
2                 NaN       NaN                    NaN           NaN   
3                 NaN       NaN                    NaN           NaN   
4                 NaN       NaN                    NaN           NaN   

  disable_comments imp_pixel  post_hint  preview      edscoreited  
0              NaN       NaN        NaN      NaN NaN          NaN  
1              NaN       NaN        NaN      NaN NaN          NaN  
2              NaN       NaN        NaN      NaN NaN          NaN  
3              NaN       NaN        NaN      NaN NaN          NaN  
4              NaN       NaN        NaN      NaN NaN          NaN  

[5 rows x 52 columns]

In [3]:
data.columns

Index(['retrieved_on', 'ups', 'media', 'selftext', 'permalink',
       'link_flair_text', 'secure_media_embed', 'title', 'distinguished',
       'thumbnail', 'downs', 'hide_score', 'domain', 'subreddit',
       'media_embed', 'author_flair_text', 'created_utc', 'quarantine',
       'score', 'edited', 'stickied', 'created', 'from_kind', 'subreddit_id',
       'url', 'secure_media', 'from_id', 'is_self', 'author', 'from',
       'archived', 'id', 'gilded', 'link_flair_css_class', 'name', 'over_18',
       'saved', 'num_comments', 'author_flair_css_class',
       'third_party_tracking', 'promoted', 'adserver_imp_pixel',
       'adserver_click_url', 'href_url', 'third_party_tracking_2',
       'mobile_ad_url', 'disable_comments', 'imp_pixel', 'post_hint',
       'preview', '', 'edscoreited'],
      dtype='object')

In [4]:
self_data = data["is_self"].value_counts()
self_data.head()

is_self
False    2583270
True     2400743
Name: count, dtype: int64

In [5]:
#removing images
data = data[data["is_self"] == True]

In [6]:
#removing over 18
data = data[data["over_18"] == False]

In [7]:
popular_subreddits = data["subreddit"].value_counts()
popular_subreddits.head(20)

subreddit
Fireteams               156936
AskReddit               144374
leagueoflegends          45003
DestinyTheGame           41816
GlobalOffensiveTrade     35240
Showerthoughts           34401
buildapc                 27979
explainlikeimfive        20113
hearthstone              16192
friendsafari             14705
pcmasterrace             14095
GlobalOffensive          13906
trees                    13473
askscience               13168
relationships            12478
techsupport              12340
wow                      12219
DotA2                    11403
jailbreak                10178
Fitness                   9367
Name: count, dtype: int64

good subreddits (many posts)
- explainlikeimfive 
- Showerthoughts
- techsupport
- AskReddit

total = 144374+12340+34401+20113 = 211228

In [8]:
def export(to_export):
    csv_export = data[data["subreddit"] == to_export]
    csv_export = csv_export[["ups","downs","title","selftext"]]
    csv_export.to_csv(f'processed_data/{to_export}.csv', index=False)

In [9]:
subreddits = ["explainlikeimfive","Showerthoughts","techsupport","AskReddit"]

for subreddit in subreddits:
    print(f"exported {subreddit} to csv")
    export(subreddit)

exported explainlikeimfive to csv
exported Showerthoughts to csv
exported techsupport to csv
exported AskReddit to csv
